### PR TITLE
PAYARA-3883 Move the groupsearcherror to a separate logger

### PR DESCRIPTION
**Pull request by AlanRoth**

### CODE REVIEW

**Comment by AlanRoth** [comment]
jenkins test please

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/ldap/LDAPRealm.java [hunk 1]
```diff
@@ -117,6 +118,8 @@
  */
 @Service
 public final class LDAPRealm extends BaseRealm {
+    // Child logger of javax.enterprise.system.core.security 
+    protected static final Logger _groupSearchLogger = Logger.getLogger(_logger.getName() + ".ldaprealm.groupsearch");
```
Comment (jbee) [inline nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/ldap/LDAPRealm.java]: Not sure about our naming guide but I'd think we do not use underscore first in newly written code.

File: nucleus/security/core/src/test/java/com/sun/enterprise/security/auth/realm/ldap/GroupSearchLoggerTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
+/*
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+ *
+ * Copyright (c) [2018] Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (Pandrex247) [inline nucleus/security/core/src/test/java/com/sun/enterprise/security/auth/realm/ldap/GroupSearchLoggerTest.java]: https://media.giphy.com/media/GcDtLf4RAdiRG/giphy.gif

File: nucleus/security/core/src/test/java/com/sun/enterprise/security/util/LogAppender.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+
+/**
+ *
+ * @author Alan Roth
+ */
+public class LogAppender extends Handler{
```
Comment (Pandrex247) [inline nucleus/security/core/src/test/java/com/sun/enterprise/security/util/LogAppender.java]: Missing a space before the `{`

File: nucleus/security/core/src/test/java/com/sun/enterprise/security/auth/realm/ldap/GroupSearchLoggerTest.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        logger = LDAPRealm._groupSearchLogger;
+        logger.addHandler(logAppender);
+    }
+    
+    @Test
+    public void groupSearchErrorLoggingTest() {
+        logger.log(INFO, "ldaprealm.groupsearcherror");
+        assertNotNull(logAppender.getLogs());
+        assertEquals(1, logAppender.getLogs().size());
+        assertEquals("ldaprealm.groupsearcherror", logAppender.getLogs().get(0).getMessage());
```
Comment (jbee) [inline nucleus/security/core/src/test/java/com/sun/enterprise/security/auth/realm/ldap/GroupSearchLoggerTest.java]: This test doesn't test the code changes but properties of the logging framework which hopefully already has that covered. Looking at the change I don't think there is a meaningful test other than making sure that the `LDAPRealm` service actually uses the group logger that you wanted it to use. I think you could just do a `new LDAPRealm()`, grab the logger field from it and check its name against your expectations.

File: nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/ldap/LDAPRealm.java (hunk 1 above)
Comment (AlanRoth) [reply nucleus/security/core/src/main/java/com/sun/enterprise/security/auth/realm/ldap/LDAPRealm.java]: Ok, I wasn't sure, I was just copying the name style of the logger it replaced

**Comment by AlanRoth** [comment]
jenkins test please

**Comment by Pandrex247** [comment]
Jenkins test please

**Pull request merged**